handlers/assign_reviewer/__init__.py

```diff
@@ -35,8 +37,19 @@ def choose_reviewer(api, pull_number):
 
     return collaborators[pull_number % len(collaborators)]
 
-welcome_msg = "Thanks for the pull request, and welcome! The Servo team is excited to review "\
-              "your changes, and you should hear from @%s (or someone else) soon."
+def check_approval(api, payload):
+    user = payload['comment']['user']['login']
+    comment = payload['comment']['body']
+    collaborators = get_collaborators(api)
+    approval = re.search(r'.*@bors-servo[: ]*r([\+=])([a-zA-Z0-9\-]*)', comment)
+
+    if approval and user in collaborators:
+        if approval.group(1) == '=':        # check if it's something like "r=username"
+            reviewer = approval.group(2)
+            if reviewer in collaborators:
+                api.set_assignee(reviewer)
+        api.set_assignee(user)      # fall back and assign the approver
```

Comment:
This line assigns `user` even if `reviewer` has been assigned on the line before. And later, if `find_reviewer` finds something, it is going to change assignee for a third time. Wouldn't it be better to somehow merge this logic so only one `set_assignee` per new comment is called?